lib/time/time.go

```diff
@@ -0,0 +1,548 @@
+		case starlark.Int:
+			i, ok := y.Int64()
+			if !ok {
+				return nil, fmt.Errorf("int value out of range (want signed 64-bit value)")
+			}
+			return Duration(x - time.Duration(i)), nil
+		case Time:
+			// duration - time = duration
+			return nil, nil
+		}
+
+	case syntax.SLASH:
+		switch y := yV.(type) {
+		case Duration:
+			if int64(y) == 0 {
+				return nil, fmt.Errorf("%s division by zero", d.Type())
+			}
+			return d / y, nil
+		case starlark.Int:
+			i, ok := y.Int64()
+			if !ok {
+				return nil, fmt.Errorf("int value out of range (want signed 64-bit value)")
+			}
+			if int64(i) == 0 {
+				return nil, fmt.Errorf("%s division by zero", d.Type())
+			}
+			return Duration(d / Duration(i)), nil
+		case Time:
+			// y := time.Time(yV.(Time))
+			// switch op {
+			// case syntax.PLUS:
+			// 	// duration + time = time
+			// 	return Time(y.Add(x)), nil
+			// case syntax.MINUS:
+			// 	// duration - time = duration
+			// 	return nil, nil
+			// }
+		}
+
+		// if int64(y) == 0 {
+		// 	return nil, fmt.Errorf("%s division by zero", d.Type())
+		// }
+		// return Duration(x / y), nil
+
+	case syntax.SLASHSLASH:
+		switch y := yV.(type) {
+		case Duration:
+			if int64(y) == 0 {
+				return nil, fmt.Errorf("%s division by zero", d.Type())
+			}
+			return d / y, nil
+		case starlark.Int:
+			i, ok := yV.(starlark.Int).Int64()
+			if !ok {
+				return nil, fmt.Errorf("int value out of range (want signed 64-bit value)")
+			}
+			return d / Duration(i), nil
+		case Time:
+			// y := time.Time(yV.(Time))
+			// switch op {
+			// case syntax.PLUS:
+			// 	// duration + time = time
+			// 	return Time(y.Add(x)), nil
+			// case syntax.MINUS:
+			// 	// duration - time = duration
+			// 	return nil, nil
+			// }
+		}
+
+		// if int64(y) == 0 {
+		// 	return nil, fmt.Errorf("time.duration floored divison by zero")
+		// }
+		// return Duration(x / y), nil
+
+	case syntax.STAR:
+		switch y := yV.(type) {
+		case Duration:
+			return d * y, nil
+		case starlark.Int:
+			i, ok := y.Int64()
+			if !ok {
+				return nil, fmt.Errorf("int value out of range (want signed 64-bit value)")
+			}
+			return d * Duration(i), nil
+		case Time:
+			// y := time.Time(yV.(Time))
+			// switch op {
+			// case syntax.PLUS:
+			// 	// duration + time = time
+			// 	return Time(y.Add(x)), nil
+			// case syntax.MINUS:
+			// 	// duration - time = duration
+			// 	return nil, nil
+			// }
+		}
+
+		// return Duration(x * y), nil
+	}
+
+	return nil, nil
+}
+
+// Time is a starlark representation of a point in time.
+type Time time.Time
+
+func newTime(thread *starlark.Thread, _ *starlark.Builtin, args starlark.Tuple, kwargs []starlark.Tuple) (starlark.Value, error) {
+	var (
+		year, month, day, hour, min, sec, nsec int
+		loc                                    string
+	)
+	if err := starlark.UnpackArgs("time", args, kwargs, "year", &year, "month", &month, "day", &day, "hour", &hour, "min", &min, "sec", &sec, "nsec", &nsec, "loc", &loc); err != nil {
```

Comment:
Maybe make all args optional to allow for a "null-time".
```suggestion
	if err := starlark.UnpackArgs("time", args, kwargs, "year?", &year, "month?", &month, "day?", &day, "hour?", &hour, "minute?", &min, "second?", &sec, "nanosecond?", &nsec, "location?", &loc); err != nil {
```